### ISSUE TITLE
Empty file on module build

### ISSUE THREAD

**Issue by mb1069** [ctSkennerton/kegg_hmm_models#1]
Hi,
I'm running into some issues when attempting to build the following modules;
M00009, M00023,  M00026, M00087, M00089, M000173, M000308, M000362, M000611 -> M00614, M000616->M000618, M000633.

```
make kegg_modules/M00611/models.hmm.h3p
mkdir kegg_modules/M00611
cat  > kegg_modules/M00611/models.hmm
rm kegg_modules/M00611/models.hmm.*
rm: cannot remove â€˜kegg_modules/M00611/models.hmm.*â€™: No such file or directory
make: [kegg_modules/M00611/models.hmm.h3p] Error 1 (ignored)
hmmpress kegg_modules/M00611/models.hmm

Error: File format problem in trying to open HMM file kegg_modules/M00611/models.hmm.
File exists, but appears to be empty?

make: *** [kegg_modules/M00611/models.hmm.h3p] Error 1
rm kegg_modules/M00611/models.hmm
```

I believe this may be due to data being unavailable from the KEGG API, but was wondering if you had a workaround for this?

Kindest,
Miguel Boland

**Comment by ctSkennerton** [comment]
yeah, I use the KEGG API to pull out the KO values for the module, but that module doesn't have any KO associated with it. It looks like M00611 is a "supermodule" that contains other modules (https://www.genome.jp/kegg-bin/show_module?M00611). You can make hmm files for the constituent modules instead.

**Comment by mb1069** [comment]
Ah fantastic!
Do you know where I could find a list of supermodules?

**Comment by ctSkennerton** [comment]
I'm afraid not. I've never systematically gone through all of the kegg modules - I didn't even realize that KEGG had modules containing other modules until your bug report.

**Comment by mb1069** [comment]
Hi again,
I've encountered another issue while building some modules; it seems that hmmpress finds colliding keys in the final steps of the process? 
The error is :
hmmpress kegg_modules/M00009/models.hmm
Working...    SSI index construction failed:
  primary keys not unique: 'sdhD,' occurs more than once
make: *** [kegg_modules/M00009/models.hmm.h3p] Error 1 

The full log can be found here (big file warning):
https://gist.github.com/mb1069/e768f3e13ade34fd3afeec48127747d7
